src/mmj/lang/Formula.java

```diff
@@ -508,6 +474,53 @@ public Sym setTyp(final Map<String, Sym> symTbl, final String typS)
     }
 
     /**
+     * Determine count of left (+1) and right (-1) parens.
+     * Ideally the value will be 0 (balanced).
+     * Note that if the value is ever negative, this returns -1 immediately -- because there's already an error.
+     * @return int parenthesis count
+     */
+    public int countBalancedParentheses() {
+        int balanceCount = 0;
+        for (Sym symbol : sym) {
+            String symbolId = symbol.getId();
+            if ("(".equals(symbolId)) {
```

Comment:
Yep, you're right; I didn't think of other symbols, but yes, it would be the same issue. I will close the PR and look at the grammar creation code.